master/internal/experiment/bulk_action.go

```diff
@@ -148,8 +151,15 @@ func editableExperimentIds(ctx context.Context, inputExpIDs []int32,
 		Model(&expIDs).
 		Column("e.id").
 		Join("JOIN projects p ON e.project_id = p.id").
-		Where("NOT e.archived").
-		Where("e.id IN (?)", bun.In(experimentIDList))
+		Where("NOT e.archived")
+
+	if filters == nil && len(experimentIDList) == 1 {
+		query = query.Where("e.id = ?", experimentIDList[0])
+	} else if filters == nil {
+		query = query.
+			Where("e.id IN (?)", bun.In(experimentIDList)).
+			Where("e.project_id = ?", inputProjectID)
+	}
```

Comment:
Yep, I agree, this is a niche way to handle this. While it had been the way this had been handled prior, I don't think that justifies keeping it this way. 

I was originally of the mindset to remove the option to specify a filter in the first place and I had began doing so; however, it proved to be a larger level of effort that what I'm comfortable taking on during a release. 

I'm not against the idea of having the behavior be intersection of the provided ids & filters. My only concern is that I'm not entirely clear on what the use case here would be for this behavior. 

I could see a case where someones select X experiments adds a filter & then performs a bulk action; and instead of use updating the set of selected experiment ids, we instead just pass in the full set of ids & filters. That case feels a little off to me due to the potential for the set of selected experiment ids being incorrect.

Reply:
In general, it's a [code smell](https://en.wikipedia.org/wiki/Code_smell) to me when functions only pay attention to their parameters some of the time. If we want something that acts based on filters and something that acts based on ids, that would ideally be two functions in my opinion—one which only accepts filters and the other that only accepts ids. If nothing else, I think the complex behavior of "ids will be ignored if filters are passed in" should be documented in the comment for the function, but ideally people don't need to read those to predict the behavior of a function (or read the function's code itself).

Reply:
I don't disagree, I think ideally we'd have separate functions to handle these workflows. In an effort to get this over the line, I'll include a comment defining this behavior and a ticket to follow up on this so that we can improve our code hygiene here.

Reply:
Sounds totally acceptable. Thanks!